src/modules/map/components/toolbox/TimeSlider.vue

```diff
@@ -1,3 +1,305 @@
+    }
+    return yearsInCommon
+})
+
+watch(screenWidth, (newValue) => {
+    setSliderWidth(newValue)
+})
+
+watch(layersWithTimestamps, (newLayers) => {
+    if (newLayers.length > 1) {
+        // checking if all layers can be set to the current year
+        if (
+            !newLayers
+                .map((layer) => layer.timeConfig.years.includes(currentYear.value))
+                .reduce((a, b) => a && b, true)
+        ) {
+            setCurrentYearAndDispatchToStore(
+                findMostRecentCommonYear(newLayers.map((layer) => layer.timeConfig))
+            )
+        }
+    } else {
+        // only one layer left, checking that it can comply with the current year, otherwise
+        // we take the most recent instead
+        const [onlyLayerLeft] = newLayers
+        if (!onlyLayerLeft.timeConfig.years.includes(currentYear.value)) {
+            setCurrentYearAndDispatchToStore(onlyLayerLeft.timeConfig.years[0])
+        }
+    }
+})
+
+// we can't watch currentYear and dispatch changes to the store here, otherwise the store gets
+// dispatch too many times when the user is moving the time slider (we wait for mouseup our
+// touchend to commit the change)
+
+onMounted(() => {
+    let initialYear
+    setSliderWidth()
+    // let's define the current year to apply to all (future) layer added to this time slider
+    const [firstLayer] = layersWithTimestamps.value
+    if (layersWithTimestamps.value.length === 1) {
+        // if there is only one layer, and its current timestamp is a valid year, we take it
+        if (firstLayer.timeConfig.currentYear !== YEAR_TO_DESCRIBE_ALL_OR_CURRENT_DATA) {
+            initialYear = firstLayer.timeConfig.currentYear
+        } else {
+            // otherwise take the first year that is not ALL_YEARS
+            initialYear = firstLayer.timeConfig.years.find(
+                (year) => year !== YEAR_TO_DESCRIBE_ALL_OR_CURRENT_DATA
+            )
+        }
+    } else {
+        // if multiple layers are visible, we need to find the closest year (from now) that is
+        // a common year between all layers
+        initialYear =
+            findMostRecentCommonYear(layersWithTimestamps.value.map((layer) => layer.timeConfig)) ||
+            YOUNGEST_YEAR
+    }
+    // We always need to propagate the changes to the store in order to have a proper time
+    // slider toggling
+    setCurrentYearAndDispatchToStore(initialYear)
+})
+
+onUnmounted(() => {
+    store.dispatch('clearPreviewYear', dispatcher)
+})
+
+function setCurrentYearAndDispatchToStore(year) {
+    currentYear.value = year
+    store.dispatch('setPreviewYear', { year: currentYear.value, ...dispatcher })
+}
+
+function setSliderWidth() {
+    // 19px of padding (7.5 on both side of the container with p-2 class and 4 with px-1)
+
+    sliderWidth.value = sliderContainer.value.clientWidth - 19 - PLAY_BUTTON_SIZE
+}
+
+function positionNodeLabel(year) {
+    const timestampIndex = ALL_YEARS.indexOf(year) || 1
```

Comment:
if you see this kinds of "default assign" through `||` you can translate them into newer JS with `??` operator instead